Classes/Systems/GithubClient.swift

```diff
@@ -164,4 +210,43 @@ struct GithubClient {
         })
     }
 
+    static func url(baseURL: String = "https://github.com/", path: String) -> URL? {
+        return URL(string: "\(baseURL)\(path)")
+    }
+
+    // MARK: Private
+    private func constructURL(from request: Request) -> String {
+        let baseURL: String
+        if let enterpriseURL = userSession?.enterpriseURL {
+            switch request.type {
+            case .api:
+                baseURL = "\(enterpriseURL)api/v3"
+            case .site:
+                baseURL = enterpriseURL
+            case .status:
+                assert(false, "Status checking is unsupported in GitHub Enterprise")
+                baseURL = "invalid-url"
```

Comment:
Not sure how we would want to make this easier to bubble up? Maybe we can handle with a Toast somewhere?